File: alphaProjectControleDeEstoque/src/main/java/com/project/conrtoleEstoque/model/Product.java
package com.project.conrtoleEstoque.model;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
public class Product {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String name;
	private Integer barCode;
	private Double price;
	private Integer quantity;
	
	//@ManyToOne(cascade = CascadeType.ALL)
	//private Category category;
}

File: estoque/src/types/Product.ts
export type Product = {
    id?: number;
	name: string;
	barCode: number;
	price: number;
	quantity: number;
    category?: Category;	
}

export type ProductPage = {
    content?: Product[];
    last: boolean;
    totalPages: number;
    totalElements: number;
    size?: number;
    number: number;
    first: boolean;
    numberOfElements?: number;
    empty?: boolean;
}

export type Category = {
    id?: number;
	name: string;
}